hare/algorithm.go

```diff
@@ -198,14 +198,20 @@ func (proc *ConsensusProcess) eventLoop() {
 		log.Int("Hare-N", proc.cfg.N), log.Int("f", proc.cfg.F), log.String("duration", (time.Duration(proc.cfg.RoundDuration)*time.Second).String()),
 		log.LayerId(uint64(proc.instanceId)), log.Int("exp_leaders", proc.cfg.ExpectedLeaders), log.String("current_set", proc.s.String()))
 
-	// set pre-round InnerMsg and send
-	builder, err := proc.initDefaultBuilder(proc.s)
-	if err != nil {
-		proc.Error("init default builder failed: %v", err)
-		return
+	// check participation
+	if !proc.shouldParticipate() {
+		proc.With().Info("Should not participate", log.Int32("round", proc.k),
```

Comment:
move this log print into `shouldParticipate`, it will spare you from writing it over and over

Reply:
Done